### PR TITLE
Marquee select in maps

### PR DESCRIPTION
**Pull request by bfinzer**
[#187255353] Feature: A marquee select tool in maps allows users to select points

* Copy marquee icons from V2 to V3
* Add a `MapMarqueeSelectButton` component to `CodapMap`
* Made `Background` accessible to both graph and map
* Get rid of the assumption that there is at most one PixiPoints. In the graph, this has fairly wide-reaching ramifications at the model, component and controller levels
* Passing a reference to an *array* of PixiPoints to `Background` so that it can build its tree from all the points in the graph or map
* In many places where we formerly had to dereference to get to the desired PixiPoints, we can now just use an instance
* Fix the bug whereby a click on the map doesn't deselect cases by installing a click handler in the leafletMap
* Move plotBackgroundColor, isTransparent and marqueeMode up to `DataDisplayContentModel` so they can be accessed by `Background` when it is installed in a map component
* In `CodapMap` render the `Background` when the map model's marqueeMode is 'selected'
* Allow clients of `MapContentModel` to set a flag, `deselectionIsDisabled` whose purpose is to disable temporarily the carrying out of `deselectAllCases`
* Added `datasetID` as a property of `IPixiPointMetadata` so that `Background` can deal with displayed points in the map where they come from different datasets
* Cleaned up the marquee select button, including using svg instead of png
* Skipping problematic jest test

### CODE REVIEW

File: v3/src/components/map/components/map-marquee-select-button.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    else {
+      mapModel.setMarqueeMode('unclicked')
+    }
+  }
+
+  if (atLeastOnePointsLayerIsVisible()) {
+    const className = mapModel.marqueeMode === 'selected' ? 'map-marquee-select selected' : 'map-marquee-select'
+    return (
+      <Portal containerRef={mapRef}>
+        <Button className={className} data-testid="map-marquee-select" aria-label='Marquee Select'
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/map/components/map-marquee-select-button.tsx:34]: I also added a tooltip

File: v3/src/components/map/components/map-point-layer.tsx [hunk 2]
```diff
@@ -243,6 +247,13 @@ export const MapPointLayer = function MapPointLayer(props: {
     )
   }, [callMatchCirclesToData, mapLayerModel])
 
+  useEffect(function _stashPixiPoints() {
+    if (pixiPointsRef.current) {
+      stashPixiPoints(pixiPointsRef.current, mapLayerModel.layerIndex)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])  // only run once on mount
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/map/components/map-point-layer.tsx:260]: Moving it up to the spot where the PixiPoints is created worked and revealed the need to define stashPixiPoints as a callback so it wouldn't be different on each render.

File: v3/src/components/map/components/map-point-layer.tsx [hunk 3]
```diff
@@ -22,11 +22,13 @@ import {useMapModelContext} from "../hooks/use-map-model-context"
 import {IMapPointLayerModel} from "../models/map-point-layer-model"
 import {MapPointGrid} from "./map-point-grid"
 
-export const MapPointLayer = function MapPointLayer(props: {
+interface IProps {
   mapLayerModel: IMapPointLayerModel
-}) {
-  const {mapLayerModel} = props,
-    {dataConfiguration, pointDescription} = mapLayerModel,
+  stashPixiPoints: (pixiPoints: PixiPoints, layerIndex: number) => void
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/map/components/map-point-layer.tsx:27]: Agreed.

**Review by kswenson — APPROVED** [review]
👍 Changes look good

**Pull request merged**